tests/test_esm_parser/test_provenance.py

```diff
@@ -164,122 +150,172 @@ def test_extract_dict():
-                                            'line': 27,
-                                            'col': 17,
-                                            'yaml_file': file_path,
-                                            'category': 'None'}
-                                    }}]}}]}}
+    file_path = pathlib.Path(str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml")))
+    check_prov = [{'line': 15, 'col': 19, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml").resolve()), 'category': 'runscript'}, {'line': 15, 'col': 22, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml").resolve()), 'category': 'runscript'}, {'line': 15, 'col': 25, 'yaml_file': str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml").resolve()), 'category': 'runscript'}]
 
     with open(file_path, "r") as file:
         esm_tools_loader.set_filename(file_path)
         data, data2 = esm_tools_loader.load(file)
 
-    assert data2 == provenance
+    config = provenance.DictWithProvenance(data, data2)
+    assert config["person"]["my_other_list"].get_provenance() == check_prov
 
 
-# Test 7 (reset the provenance of a leaf)
-def test_get_provenance_7():
-    config_fesom = provenance.DictWithProvenance({"fesom": {"update_test": True}}, "new_provenance")
+# Test 12: Check set_provenance of a list entry
+def test_check_set_provenance_list():
+    os.environ['USER'] = "some_user"
+    esm_tools_loader = yaml_to_dict.EsmToolsLoader()
+    file_path = pathlib.Path(str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml")))
+    new_prov = {'line': 15, 'col': 25, 'yaml_file': 'example.yaml', 'category': 'from_a_list'}
+    check_prov = [new_prov, new_prov, new_prov]
 
-    config["fesom"].update(config_fesom["fesom"])
+    with open(file_path, "r") as file:
+        esm_tools_loader.set_filename(file_path)
+        data, data2 = esm_tools_loader.load(file)
 
-    check_provenance = {
-        "echam": {
-            "type": "a_string",
-            "files": {
-                "greenhouse": {"kind": "a_new_string", "path_in_computer": "a_string"}
-            },
-        },
-        "fesom": {"asd": 2, "model": 2, "update_test": "new_provenance"},
-        "computer": None,
-        True: None,
-    }
-    assert config.get_provenance() == check_provenance
+    config = provenance.DictWithProvenance(data, data2)
+    config["person"]["my_other_list"].set_provenance(new_prov)
+    assert config["person"]["my_other_list"].get_provenance() == check_prov
+
+
+# Test 13: Check set_provenance of a single list entry
+def test_check_set_provenance_of_single_list_entry():
+    os.environ['USER'] = "some_user"
+    esm_tools_loader = yaml_to_dict.EsmToolsLoader()
+    file_path = pathlib.Path(str(pathlib.Path(f"{os.environ['GITHUB_WORKSPACE']}/tests/test_esm_parser/example.yaml")))
```

Comment:
same as above with the locally run tests

Reply:
Done